apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts

```diff
@@ -564,3 +468,159 @@ function DeleteQuestionModal({
+        <div class="col-1">
+          <div class="badge color-green3">Public</div>
+        </div>
+        <div class="col-auto">This question is publicly shared.</div>
+      </div>
+    `;
+  }
+
+  return html`
+    ${sharingSetsIn.length === 0
+      ? html`<small class="text-muted px-3">This question is not being shared</small>`
+      : html`
+          <small class="text-muted"
+            >Shared With:
+            ${sharingSetsIn.map(function (sharing_set) {
+              return html` <span class="badge color-gray1"> ${sharing_set?.name} </span> `;
+            })}
+          </small>
+        `}
+    ${hasCoursePermissionOwn
+      ? html`
+          ${sharingSetsOther.length > 0
+            ? html`
+                <form name="sharing-set-add" method="POST" class="d-inline">
+                  <input type="hidden" name="__action" value="sharing_set_add" />
+                  <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+                  <div class="btn-group btn-group-sm" role="group">
+                    <button
+                      id="addSharingSet"
+                      type="button"
+                      class="btn btn-sm btn-outline-dark dropdown-toggle"
+                      data-toggle="dropdown"
+                      aria-haspopup="true"
+                      aria-expanded="false"
+                    >
+                      Add...
+                    </button>
+                    <div class="dropdown-menu" aria-labelledby="addSharingSet">
+                      ${sharingSetsOther.map(function (sharing_set) {
+                        return html`
+                          <button
+                            class="dropdown-item"
+                            name="unsafe_sharing_set_id"
+                            value="${sharing_set.id}"
+                          >
+                            ${sharing_set.name}
+                          </button>
+                        `;
+                      })}
+                    </div>
+                  </div>
+                </form>
+              `
+            : ''}
+          <button
+            class="btn btn-sm btn-outline-primary"
+            type="button"
+            data-toggle="modal"
+            data-target="#publiclyShareModal"
+          >
+            Share Publicly
+          </button>
+          ${PubliclyShareModal({
+            csrfToken,
+            qid,
+          })}
+        `
+      : ''}
+  `;
+}
+
+function TagRows({ tags }) {
+  if (!tags || tags.length === 0) {
+    return html` <small class="text-muted px-3"> This question does not have any tags. </small>`;
+  }
+  return tags.map((tag) => {
+    return html`
+      <div class="row px-3">
+        <div class="col-1">
+          <span class="badge color-${tag.color}"> ${tag.name} </span>
+        </div>
+        <div class="col-auto">${tag.description}</div>
+      </div>
+    `;
+  });
+}
+
+function AssessmentRows({ assessmentsWithQuestion }) {
```

Comment:
Some weird stuff happens on narrow viewports:

<img width="342" alt="Screenshot 2024-04-19 at 14 11 42" src="https://github.com/PrairieLearn/PrairieLearn/assets/1476544/84470277-0704-40f9-b024-d42581a5e85f">

I'm honestly wondering if we need to show this here; it seems strange to show this on a "settings" page? One can already see all assessments on which a question is used on the "statistics" question tab. cc @mwest1066 @mfsilva22 for input: would you be sad if the assessments list on the question settings page went away?

Reply:
It wouldn't be terrible to remove it, but I think it makes sense to have it here. I don't think that we need to show titles for the assessments, if that's what's causing the problems? At the moment we just show a nice compact list like this:
<img width="231" alt="image" src="https://github.com/PrairieLearn/PrairieLearn/assets/875833/d07556c3-f2dd-4763-8726-139c7346914e">

Reply:
I updated to show the compact list by course instance in [381d728](https://github.com/PrairieLearn/PrairieLearn/pull/9687/commits/381d7287d02a906cb1405663be90d4f84dc38348). Would something like this make sense? 

![Screenshot 2024-04-24 at 12 03 27 PM](https://github.com/PrairieLearn/PrairieLearn/assets/114321556/96bee2d2-73b4-45fb-9929-9f5243b9bfee)

and on narrower viewports:
![Screenshot 2024-04-24 at 12 03 51 PM](https://github.com/PrairieLearn/PrairieLearn/assets/114321556/0ec155fc-c52b-40cc-ad30-07f64129882a)

We could also decide which content is hidden or displayed on certain size viewports by using Bootstraps built-in breakpoints (https://getbootstrap.com/docs/4.0/utilities/display/). This could also be useful for deciding if we render the descriptions in the topic and tags sections as well.